app/Shop/Orders/Repositories/OrderRepository.php

```diff
@@ -207,4 +209,22 @@ public function buildOrderDetails(Collection $items)
             }
         });
     }
+
+    /**
+     * @return Collection $addresses
+     */
+    public function getAddresses() : Collection
+    {
+        $addresses = $this->model->first()->address()->get();
+        return $addresses;
```

Comment:
Not sure why you want the always the first order? When you pass the order object in the constructor, you already specifying the "order" and get it's address right? So , you just need to:

```
$orderRepository = new OrderRepository($order);
$addresses = $orderRepository->getAddresses();

// Get the first address in the collection
$addresses->first();
```
```

Reply:
@Daniyal-Javani You can now already return it without any variable assignment

```
return $this->model->address()->get();

// instead of doing
 $addresses = $this->model->address()->get();
 return $addresses;
```